scripts-endpoint.md

```diff
@@ -0,0 +1,169 @@
+{
+   "name" : "import",
+   "description" : "Import items into DSpace",
+   "parameters" : [
+    {
+      "name" : "a",
+      "description" : "add items to DSpace"
+      "type" : "boolean"
+    },
+    {
+      "name" : "b",
+      "description" : "add items to DSpace via Biblio-Transformation-Engine (BTE)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "c",
+      "description" : "destination collection(s) database ID"
+      "type" : "id"
+    },
+    {
+      "name" : "d",
+      "description" : "delete items listed in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "e",
+      "description" : "email of eperson doing importing"
+      "type" : "string"
+    },
+    {
+      "name" : "i",
+      "description" : "input type in case of BTE import"
+      "type" : "string"
+    },
+    {
+      "name" : "n",
+      "description" : "if sending submissions through the workflow, send notification emails"
+      "type" : "boolean"
+    },
+    {
+      "name" : "p",
+      "description" : "apply template"
+      "type" : "string"
+    },
+    {
+      "name" : "q",
+      "description" : "don't display metadata"
+      "type" : "boolean"
+    },
+    {
+      "name" : "m",
+      "description" : "mapfile items in mapfile"
+      "type" : "file"
+    },
+    {
+      "name" : "r",
+      "description" : "replace items in mapfile"
+      "type" : "boolean"
+    },
+    {
+      "name" : "R",
+      "description" : "resume a failed import (add only)"
+      "type" : "boolean"
+    },
+    {
+      "name" : "t",
+      "description" : "test run - do not actually import items"
+      "type" : "boolean"
+    },
+    {
+      "name" : "w",
+      "description" : "send submission through collection's workflow"
+      "type" : "boolean"
+    },
+    {
+      "name" : "z",
+      "description" : "zip file containig the import"
+      "type" : "file"
+    }
+   ]
+}
+```
+
+Following parameter types are available:
+* `string`: A regular string value
+* `date`: A string that has a valid date pattern
+* `boolean`: true or false
+* `file`: For parameters with this type, the user has to provide a file
+* `output`: Parameters with this type define the name of the output file. This name can be used later to download the the output file (e.g. when running `export` or `metadata-export`).
+
+### Script Invocation
+**POST /api/admin/scripts/<:script-name>**
+
+POST requests to this endpoint will start the corresponding script with the provided parameters. All parameter values should be provided in the body that has to use the `multipart/form-data` content type. Once the upload is complete and the script was started successfully, this endpoint will return details on the scripts execution
+
+The POST request body can contain the optional parameter `startTime`. When this parameter contains a valid timestamp, the script will start at that timestamp.
+
+```
+{
+  "processId" : "000003f1-a850-49de-af03-997272d834c9",
+  "scriptName" : "import",
```

Comment:
you should have a link to the script that have generated the process instead of the name.
You can add an attribute to list the parameters used for the execution

Reply:
I'll try to add this to the contract.